packages/theme/src/provider.tsx

```diff
@@ -1,23 +1,60 @@
-import { FC } from 'react'
-import {
+import React, { FC, PropsWithChildren, useEffect } from 'react'
+import styled, {
   ThemeProvider as StyledThemeProvider,
   ThemeProviderProps,
 } from 'styled-components'
-import { themeDefault, themeDark, themeLight } from './themes'
+import { themeDark, themeDefault, themeLight } from './themes'
 import { Theme } from './types'
+import { initColors } from './document-head-contents'
+import { generateCssColorVariables } from './document-head-contents/utils/generate-css-color-variables'
+import { useThemeToggle } from '@lidofinance/cookie-theme-toggler'
 
-export const ThemeProvider: FC<ThemeProviderProps<Theme>> = (props) => {
-  const { theme = themeDefault, ...rest } = props
+// you need this if you're using themes, so we initiate this automatically
+initColors()
 
-  return <StyledThemeProvider theme={theme} {...rest} />
+const StyledWrapper = styled.div<{ colors?: Record<string, string> }>`
+  display: contents;
+  ${({ colors }) => (colors ? generateCssColorVariables(colors) : null)}
```

Comment:
Why do we need it here if we set them in header?

Reply:
outdated